clumper/clump.py

```diff
@@ -174,19 +188,102 @@ def read_csv(cls, path, delimiter=",", fieldnames=None, n=None):
             if n <= 0:
                 raise ValueError("Number of lines to read must be > 0.")
 
-        if path.startswith("https:") or path.startswith("http:"):
+        # conveniently excludes pathlib files here and removes
+        # the need to write code to check pathlib files in other places.
+        if isinstance(path, str) and path.startswith(("https:", "http:")):
             with urllib.request.urlopen(path) as resp:
                 if fieldnames is None:
                     fieldnames = resp.readline().decode().strip().split(",")
-                # this section allows us to chunk the rows, if nrows is supplied
+                # This section allows us to chunk the rows, if nrows is supplied.
                 body = it.islice(resp, 0, n)
                 body = (word.decode().strip().split(",") for word in body)
                 body = it.product([fieldnames], body)
-                return Clumper([dict(zip(key, values)) for key, values in body])
+                result = [dict(zip(key, values)) for key, values in body]
+        else:
+            with open(path, newline="") as csvfile:
+                reader = csv.DictReader(
+                    csvfile, delimiter=delimiter, fieldnames=fieldnames
+                )
+                # python version less than 3.8 returns an OrderedDict
+                result = [dict(entry) for entry in it.islice(reader, 0, n)]
+
+        # null values, same as missing keys.
+        # if there are null values/missing keys, they will be truncated from the dictionary.
```

Comment:
Nitpick. Punctuation is missing -> needs to start with captical letters.

Reply:
Hahaha. Sure. I'll fix it.

Reply:
@koaning made the punctuation updates